cmd/manager/main.go

```diff
@@ -259,46 +253,72 @@ func addMetrics(ctx context.Context, cfg *rest.Config) error {
 		log.Info("Could not create metrics Service", "error", err.Error())
 	}
 
-	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
-	// necessary to configure Prometheus to scrape metrics from this operator.
 	services := []*v1.Service{service}
+	mclient := monclientv1.NewForConfigOrDie(cfg)
+	copts := metav1.CreateOptions{}
 
-	// The ServiceMonitor is created in the same namespace where the operator is deployed
-	_, err = metrics.CreateServiceMonitors(cfg, operatorNs, services)
-	if err != nil {
-		log.Info("Could not create ServiceMonitor object", "error", err.Error())
-		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
-		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
-		if err == metrics.ErrServiceMonitorNotPresent {
-			log.Info("Install prometheus-operator in your cluster to create ServiceMonitor objects", "error", err.Error())
+	for _, s := range services {
+		if s == nil {
+			continue
 		}
-
+		sm := GenerateServiceMonitor(s)
+
+		// ErrSMMetricsExists is used to detect if the -metrics ServiceMonitor already exists
+		var ErrSMMetricsExists = fmt.Sprintf("servicemonitors.monitoring.coreos.com \"%s-metrics\" already exists", muocfg.OperatorName)
+
+		log.Info(fmt.Sprintf("Attempting to create service monitor %s", sm.Name))
+		// TODO: Get SM and compare to see if an UPDATE is required
+		_, err := mclient.ServiceMonitors(operatorNs).Create(ctx, sm, copts)
+		if err != nil {
+			if err.Error() != ErrSMMetricsExists {
+				return err
+			}
+			log.Info("ServiceMonitor already exists")
+		}
+		log.Info(fmt.Sprintf("Successfully configured service monitor %s", sm.Name))
 	}
 	return nil
 }
 
-// serveCRMetrics gets the Operator/CustomResource GVKs and generates metrics based on those types.
-// It serves those metrics on "http://metricsHost:operatorMetricsPort".
-func serveCRMetrics(cfg *rest.Config, operatorNs string) error {
-	// The function below returns a list of filtered operator/CR specific GVKs. For more control, override the GVK list below
-	// with your own custom logic. Note that if you are adding third party API schemas, probably you will need to
-	// customize this implementation to avoid permissions issues.
-	filteredGVK, err := k8sutil.GetGVKsFromAddToScheme(apis.AddToScheme)
-	if err != nil {
-		return err
+// GenerateServiceMonitor generates a prometheus-operator ServiceMonitor object
+// based on the passed Service object.
+func GenerateServiceMonitor(s *v1.Service) *monitoringv1.ServiceMonitor {
```

Comment:
The same function is defined in operator-sdk, service-monitor.go

Why we cannot just use that?

Reply:
good catch, forgot to repack this after unpacking the sdk flow in main.go. 

thanks, fixed + sqaushed.